libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -208,63 +274,19 @@ bool AP_Arming::ins_checks(bool report)
         }
 
         // check all accelerometers point in roughly same direction
-        if (ins.get_accel_count() > 1) {
-            const Vector3f &prime_accel_vec = ins.get_accel();
-            for(uint8_t i=0; i<ins.get_accel_count(); i++) {
-                if (!ins.use_accel(i)) {
-                    continue;
-                }
-                // get next accel vector
-                const Vector3f &accel_vec = ins.get_accel(i);
-                Vector3f vec_diff = accel_vec - prime_accel_vec;
-                // allow for user-defined difference, typically 0.75 m/s/s. Has to pass in last 10 seconds
-                float threshold = accel_error_threshold;
-                if (i >= 2) {
-                    /*
-                      we allow for a higher threshold for IMU3 as it
-                      runs at a different temperature to IMU1/IMU2,
-                      and is not used for accel data in the EKF
-                     */
-                    threshold *= 3;
-                }
-
-                // EKF is less sensitive to Z-axis error
-                vec_diff.z *= 0.5f;
-
-                if (vec_diff.length() <= threshold) {
-                    last_accel_pass_ms[i] = AP_HAL::millis();
-                }
-                if (AP_HAL::millis() - last_accel_pass_ms[i] > 10000) {
-                    if (report) {
-                        gcs().send_text(MAV_SEVERITY_CRITICAL, "PreArm: Accels inconsistent");
-                    }
-                    return false;
-                }
+        if (!ins_accels_consistent(ins)) {
+            if (report) {
+                gcs().send_text(MAV_SEVERITY_CRITICAL, "PreArm: Accels inconsistent");
```

Comment:
It's a behavior change, but could we think about moving this inside the check so that the error message can be "PreArm: Accel %d is inconsistent" this would be great when a user wants to know which IMU is causing issues.

Reply:
I think we'd need to rejig things so we told the user which was the current primary... this is a pretty significant change, not code-wise but ui-wise.  Not saying we shouldn't - just not here.

@amilcarlucas raised some good points about the accels-consistent thing which should be addressed, and those hacks are probably not accurate on some of our more recent triple-IMU systems.

We should consider skipping checking the primary IMU against itself.

We should consider resetting `last_accel_pass_ms` and friend when we change primary-IMU.

We should not bomb out as soon as we find a failure and report all failures instead.

We should run the gyros and accels check even if other checks have failed so the "last pass" numbers get updated for them.  I actually have a PR which would do that.

We should update the results for accels and gyros we're not using, on the basis that it's less code and if the user enabled them they don't have to wait for the checks to pass.

These methods should use the ins singleton.

You don't have to go far to find more work.  This won't be my last PR.